futures-channel/src/mpsc/mod.rs

```diff
@@ -273,16 +136,15 @@ struct Inner<T> {
     state: AtomicUsize,
 
     // Atomic, FIFO queue used to send messages to the receiver
-    message_queue: Queue<Option<T>>,
-
-    // Atomic, FIFO queue used to send parked task handles to the receiver.
-    parked_queue: Queue<Arc<Mutex<SenderTask>>>,
+    message_queue: Queue<T>,
 
-    // Number of senders in existence
-    num_senders: AtomicUsize,
+    // Atomic, FIFO queue used to send wakers for blocked `Sender` tasks to the `Receiver`.
+    sender_waker_queue: Queue<Arc<Mutex<SenderWaker>>>,
+    // Lock for popping off of the sender_waker_queue
+    sender_waker_pop_lock: Mutex<()>,
```

Comment:
i don't love that this is necessary, but it shouldn't be contended often-- it's only used in receiving and in the `Drop` impl of `Sender`.